server/src/main/java/io/crate/planner/optimizer/rule/MoveOrderBeneathNestedLoop.java

```diff
@@ -96,8 +96,9 @@ public LogicalPlan apply(Order order,
             RefVisitor.visitRefs(orderExpr, gatherRelationsFromRef);
         }
         if (relationsInOrderBy.size() == 1) {
-            RelationName relationInOrderBy = relationsInOrderBy.iterator().next();
-            if (relationInOrderBy == nestedLoop.topMostLeftRelation().relationName()) {
+            var relationInOrderBy = relationsInOrderBy.iterator().next();
+            var topMostLeftRelation = nestedLoop.getRelationNames().get(0);
+            if (relationInOrderBy.equals(topMostLeftRelation)) {
```

Comment:
In a nested join, `getRelationNames()` will return a concat list of the source relations. If the left source relation returns `[t1, t2]`, `nestedLoop.getRelationNames().get(0)` will refer to `t1` ignoring any order for `t2`. Looks like an issue to me.
Whouldn't using `NestedLoopJoin.sources().get(0).getRelationNames()` and then do a `contain()` check instead solve all issues and would not rely on ordered relations?

Reply:
Isn't the top-most-left-relation in this case `t1`?

Reply:
```
NestedLoopJoin[(a = d)]                
 ├ NestedLoopJoin[(c = d)]        
 │  ├ Collect[doc.t4 | [d] | true] 
 │  └ Collect[doc.t3 | [c] | true] 
 └ NestedLoopJoin[(a = b)] 
    ├ Collect[doc.t2 | [b] | true] 
    └ Collect[doc.t1 | [a] | true] <- top most left relation
```

Reply:
and if it would be further nested this case is still valid:

```
NestedLoopJoin[(c = d)]                
 ├ NestedLoopJoin[(a = b)]        
 │  ├ Collect[doc.t1 | [a] | true] <- top most left relation
 │  ├ NestedLoopJoin[(b = c)] 
 |  ├ Collect[doc.t2 | [b] | true] 
 |  └ Collect[doc.t3 | [c] | true]    
 └ NestedLoopJoin[(d = e)] 
    ├ Collect[doc.t4 | [d] | true] 
    └ Collect[doc.t5 | [e] | true] 
````

Reply:
> Isn't the top-most-left-relation in this case t1?

Yes but what if I order on `t2`, looks like this would not be pushed down?

Reply:
I also wonder how this would currently work in nested joins, as the `topMostLeftRelation` is a `AnalyzedRelation` while `JoinRelation.getRelationName()` throws an error. That's also why I asked for a test scenario...

Reply:
`JoinRelation` is an `AnalyzedStatement` , these are LogicalPlans.

Reply:
> JoinRelation is an AnalyzedStatement , these are LogicalPlans.

The existing code uses the `AnalyzedRelation` as the `topMostLeftRelation`, so why did the rule work with nested joins and ordering?

Anyway, please add a test. Also for ordering on the 2 relation of a nested join. My whole point here is that I wonder if this solution will improve things.

Reply:
See the individual commits. i added `topMostLeftRelationName` to the nested-loop-join for testing.

Reply:
> Yes but what if I order on `t2`, looks like this would not be pushed down?

This is true, but this is another issue.